static/js/stream_sort.js

```diff
@@ -26,7 +26,8 @@ function filter_streams_by_search(streams, search_term) {
     var filtered_streams = _.filter(streams, function (stream) {
         return _.any(search_terms, function (search_term) {
             var lower_stream_name = stream.toLowerCase();
-            var cands = lower_stream_name.split(" ");
+            // This regex allows us to split at " " and "-"
+            var cands = lower_stream_name.split(/-| /);
```

Comment:
Tim, thoughts on including a bunch of things here? like `-`, `_`, `/`, `:`, etc?

Reply:
We need to be careful about characters that have other meanings.  Certainly `:` would be a bad choice in typeahead because of overlap with emoji.  I think `_- /` are the widest set that makes sense.